### PR TITLE
e2e: add WaitUntil check for ns in subnet labelSelector

### PR DESCRIPTION
**Pull request by hongzhen-ma**
# Pull Request

- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR
- Tests

## Which issue(s) this PR fixes

Fixes #(issue-number)

### CODE REVIEW

File: test/e2e/kube-ovn/subnet/subnet-selectors.go [hunk 1]
```diff
@@ -90,6 +90,13 @@ var _ = framework.Describe("[group:subnet]", func() {
...
 
 		checkNs2 := nsClient.Get(ns2Name)
+		framework.WaitUntil(time.Second, 30*time.Second, func(_ context.Context) (bool, error) {
+			checkNs2 = nsClient.Get(ns2Name)
+			if checkNs2.Annotations[util.LogicalSwitchAnnotation] == subnet.Name {
+				return true, nil
+			}
+			return false, nil
+		}, fmt.Sprintf("failed to update annotation for ns %s", checkNs2.Name))
 		framework.ExpectHaveKeyWithValue(checkNs2.Annotations, util.LogicalSwitchAnnotation, subnet.Name)
```
Comment (zhangzujian) [inline test/e2e/kube-ovn/subnet/subnet-selectors.go:100]:
```suggestion
```
这个就不需要了

File: test/e2e/kube-ovn/subnet/subnet-selectors.go [hunk 2]
```diff
@@ -99,6 +106,13 @@ var _ = framework.Describe("[group:subnet]", func() {
...
 		checkNs2 = nsClient.Get(ns2Name)
+		framework.WaitUntil(time.Second, 30*time.Second, func(_ context.Context) (bool, error) {
+			checkNs2 = nsClient.Get(ns2Name)
+			if checkNs2.Annotations[util.LogicalSwitchAnnotation] != subnet.Name {
+				return true, nil
+			}
+			return false, nil
+		}, fmt.Sprintf("failed to update annotation for ns %s", checkNs2.Name))
 		lsAnnotation := checkNs2.Annotations[util.LogicalSwitchAnnotation]
 		framework.ExpectNotEqual(lsAnnotation, subnet.Name)
```
Comment (zhangzujian) [inline test/e2e/kube-ovn/subnet/subnet-selectors.go:117]:
```suggestion
```

Comment (hongzhen-ma) [reply test/e2e/kube-ovn/subnet/subnet-selectors.go]: 已修改

File: test/e2e/kube-ovn/subnet/subnet-selectors.go (hunk 1 above)
Comment (hongzhen-ma) [reply test/e2e/kube-ovn/subnet/subnet-selectors.go]: 已修改

**Review by zhangzujian — APPROVED** [review]

**Pull request merged**